language-plutus-core/src/Language/PlutusCore/Constant/Typed.hs

```diff
@@ -336,24 +309,35 @@ makeRightReflectT = ReflectT . lift . InnerT
 -- See Note [Converting PLC values to Haskell values].
 -- Types and terms are supposed to be closed, hence no 'Quote'.
 -- | Haskell types known to exist on the PLC side.
-class KnownDynamicBuiltinType dyn where
+class KnownType a where
     -- | The type representing @dyn@ used on the PLC side.
-    toTypeEncoding :: proxy dyn -> Type TyName ()
+    toTypeAst :: proxy a -> Type TyName ()
 
     -- | Convert a Haskell value to the corresponding PLC value.
     -- 'Nothing' represents a conversion failure.
-    makeDynamicBuiltin :: dyn -> Maybe (Term TyName Name ())
+    makeKnown :: a -> Term TyName Name ()
```

Comment:
Removed the `Maybe`, we now do not have any failing conversions from Haskell to Plutus Core.